config/v1/stable.clusterversion.testsuite.yaml

```diff
@@ -14,3 +14,108 @@ tests:
       kind: ClusterVersion
       spec:
         clusterID: foo
+  - name: Should allow image to be set
+    initial: |
+      apiVersion: config.openshift.io/v1
+      kind: ClusterVersion
+      spec:
+        clusterID: foo
+        desiredUpdate:
+          image: bar
+    expected: |
+      apiVersion: config.openshift.io/v1
+      kind: ClusterVersion
+      spec:
+        clusterID: foo
+        desiredUpdate:
+          image: bar
+  - name: Should allow version to be set
+    initial: |
+      apiVersion: config.openshift.io/v1
+      kind: ClusterVersion
+      spec:
+        clusterID: foo
+        desiredUpdate:
+          version: 4.11.1
+    expected: |
+      apiVersion: config.openshift.io/v1
+      kind: ClusterVersion
+      spec:
+        clusterID: foo
+        desiredUpdate:
+          version: 4.11.1
+  - name: Should allow architecture and version to be set
+    initial: |
+      apiVersion: config.openshift.io/v1
+      kind: ClusterVersion
+      spec:
+        clusterID: foo
+        desiredUpdate:
+          architecture: Multi
+          version: 4.11.1
+    expected: |
+      apiVersion: config.openshift.io/v1
+      kind: ClusterVersion
+      spec:
+        clusterID: foo
+        desiredUpdate:
+          architecture: Multi
+          version: 4.11.1
+  - name: Should not allow architecture and image to be set
+    initial: |
+      apiVersion: config.openshift.io/v1
+      kind: ClusterVersion
+      spec:
+        clusterID: foo
+        desiredUpdate:
+          architecture: Multi
+          version: 4.11.1
+          image: bar
+    expectedError: "cannot set both Architecture and Image"
+  - name: Version must be set if architecture is set
+    initial: |
+      apiVersion: config.openshift.io/v1
+      kind: ClusterVersion
+      spec:
+        clusterID: foo
+        desiredUpdate:
+          architecture: Multi
+    expectedError: "Version must be set if Architecture is set"
+  onUpdate:
+  - name: Should not allow image to be set if architecture set
+    initial: |
+      apiVersion: config.openshift.io/v1
+      kind: ClusterVersion
+      spec:
+        clusterID: foo
+        desiredUpdate:
+          architecture: Multi
+          version: 4.11.1
+    updated: |
+      apiVersion: config.openshift.io/v1
+      kind: ClusterVersion
+      spec:
+        clusterID: foo
+        desiredUpdate:
+          architecture: Multi
+          version: 4.11.1
+          image: bar
+    expectedError: "cannot set both Architecture and Image"
+  - name: Should not allow architecture to be set if image set
+    initial: |
+      apiVersion: config.openshift.io/v1
+      kind: ClusterVersion
+      spec:
+        clusterID: foo
+        desiredUpdate:
+          image: bar
+    updated: |
+      apiVersion: config.openshift.io/v1
+      kind: ClusterVersion
+      spec:
+        clusterID: foo
+        desiredUpdate:
+          architecture: Multi
+          version: 4.11.1
```

Comment:
I'd expect us to leave off `version` for the `Should not allow architecture to be set if image set` test-case.  And I don't understand why we're using `initial`/`updated` for this test-case, but `initial`/`expected` for the earlier test-cases.  Can we continue with the `initial`/`expected` pattern, since none of our Common Expression Language (CEL) evaluation rules depend on prior state?